.github/workflows/build-push-docker.yaml

```diff
@@ -0,0 +1,49 @@
+name: Build and Push Docker Image
+
+on:
+  release:
+    types:
+      - created
+
+jobs:
+  build:
+    name: Build and push container
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Setup tag version
+        id: generate_tag
+        shell: bash
+        run: |
+          if [ "${{ github.ref_type }}" == "tag" ]; then
+            export IMAGE_TAG="${{ github.ref_name }}-$(echo $GITHUB_SHA | cut -c 1-7)"
+            echo "release=true" >> $GITHUB_OUTPUT
+          elif grep -q "main" <<< "${{ github.ref }}"; then
```

Comment:
I believe this method of tagging container images is not suitable for open-source projects.

Reply:
Can you suggest one @ducminhle ? Should we use the tag name only ( without sha ) ?

Reply:
Glad to hear your suggestion on this @ducminhle

Reply:
1. we should tag container image with tag (without commit hash) and latest for the last image.
2. we don't need pr-<commit hash>.
3. main => main-<commit hash>.